### PR TITLE
Add org configuration skeleton

### PR DESCRIPTION
**Pull request by fejta**
Implements the config structure specified in https://docs.google.com/document/d/18Kw8InxYk4OF14iTXXfvD2oy3AuoDxfqAmtvf6NbwBI/edit

Creates a skeleton peribolos binary that applies the config (will implement in subsequent PRs)

/assign @krzyzacy @cblecker @stevekuznetsov @kargakis

### CODE REVIEW

File: prow/cmd/peribolos/main.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	return nil
+}
+
+func configureTeam(client *github.Client, num int, team org.Team) error {
+	// create team if num < 0
+	// else diff and patch
+
+	// people = set(all org members)
+	// remove = people - orgConfig.members - orgConfig.admins
+	// update = oc.members - [p for p in people if member]
```
Comment (stevekuznetsov) [inline prow/cmd/peribolos/main.go]: @fejta how can we tell you think in Python? :stuck_out_tongue:

File: prow/config/org/org.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	Description                  *string              `json:"description,omitempty"`
+	Location                     *string              `json:"location,omitempty"`
+	HasOrganizationProjects      *bool                `json:"has_organization_projects,omitempty"`
+	HasRepositoryProjects        *bool                `json:"has_repository_projects,omitempty"`
+	DefaultRepositoryPermission  *RepoPermissionLevel `json:"default_repository_permission,omitempty"`
+	MembersCanCreateRepositories *bool                `json:"members_can_create_repositories,omitempty"`
+}
+
+// Org declares metadata as well as its people and teams.
+type Org struct {
```
Comment (stevekuznetsov) [inline prow/config/org/org.go]: Doesn't `go lint` complain when you don't use the package name as a namespace? Thiss could be `org.Config` and below could be `org.Metadata`

File: prow/config/org/org.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+// MashalText returns the byte representation of the permission
+func (l RepoPermissionLevel) MarshalText() ([]byte, error) {
+	return []byte(l), nil
+}
+
+// UnmarshalText validates the text is a valid string
+func (l *RepoPermissionLevel) UnmarshalText(text []byte) error {
+	switch v := RepoPermissionLevel(text); v {
+	case "read", "write", "admin", "none":
```
Comment (stevekuznetsov) [inline prow/config/org/org.go]: I'd add

```go
const (
    RepoPermissionLevelRead RepoPermissionLevel = "read"
    RepoPermissionLevelWrite                    = "write"
    RepoPermissionLevelAdmin                    = "admin"
    RepoPermissionLevelNone                     = "none"
)
```

Comment (fejta) [reply prow/config/org/org.go]: Done

File: prow/config/org/org.go (hunk 2 above)
Comment (fejta) [reply prow/config/org/org.go]: Done

File: prow/config/org/org.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+// MarshalText returns bytes that equal secret or closed
+func (p Privacy) MarshalText() ([]byte, error) {
+	return []byte(p), nil
+}
+
+// UnmarshalText returns an error if text != secret or closed
+func (p *Privacy) UnmarshalText(text []byte) error {
+	switch v := Privacy(text); v {
+	case "secret", "closed":
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/config/org/org.go]: Done

**Comment by fejta** [comment]
@cblecker @spiffxp can one/both of you take a look today?

File: prow/cmd/peribolos/main.go (hunk 1 above)
Comment (BenTheElder) [reply prow/cmd/peribolos/main.go]: :heart: list comprehensions :heart:

File: prow/cmd/peribolos/main.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	if _, err := url.Parse(o.endpoint); err != nil {
+		return fmt.Errorf("invalid --endpoint URL: %v", err)
+	}
+	return nil
+}
+
+func main() {
+	o := parseOptions()
+
+	if o.config != "TODO(fejta): implement me" {
```
Comment (BenTheElder) [inline prow/cmd/peribolos/main.go]: heh 👍

**Pull request merged**